pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1336,12 +1314,7 @@ public LLRealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableCo
     _messageDecoder = StreamDecoderProvider.create(_partitionLevelStreamConfig, fieldsToRead);
     _clientId = streamTopic + "-" + _partitionGroupId;
 
-    // Create record transformer
-    _recordTransformer = CompositeTransformer.getDefaultTransformer(tableConfig, schema);
-
-    // Create complex type transformer
-    _complexTypeTransformer = ComplexTypeTransformer.getComplexTypeTransformer(tableConfig);
-
+    _transformPipeline = new TransformPipeline(tableConfig, schema);
```

Comment:
are you also going to change `SegmentIndexCreationDriverImpl` ?

Reply:
Thanks, good pointer. Did not know there's that too. Will work on it and ping you later when ready

Reply:
Not sure why `SegmentIndexCreationDriverImpl` needs to change, but if you do, please do so in another PR (unless those changes are needed for the transform pipeline to work)

Reply:
On second thought, @npawar since that File touches even more places in Pinot, can we keep the PR small? I promise I will do a follow up PR to further tweak the `TransformPipeline` so it will be used by `SegmentIndexCreationDriverImpl` too?

Reply:
@mcvsubbu  I think we are on the same page. Those code in `SegmentIndexCreationDriverImpl` do look very similar. I will work on a second PR to reduce code waste

Reply:
@mcvsubbu `SegmentIndexCreationDriverImpl` needs to change because that's one of the 2 places where we decode + transform records. 
@dongxiaoman yes, sounds good on doing it in followup!